
fuer GCC ausprobieren:
 -gdwarf-2 

-CFLAGS=-xmodel=kernel -m64
+CFLAGS= -ffreestanding -mcmodel=kernel -mno-red-zone -m64


fuse_create_vnode

        vp->v_data = vdata;
        vdata->nodeid = nodeid;
        vdata->par_nid = parent_nid;
#ifndef DONT_CACHE_ATTRIBUTES
        vdata->uid = (uid_t)-1;
        vdata->gid = (gid_t)-1;
#endif





diff -ur illumos-fusefs-Version-1.3.1/kernel/amd64/Makefile /export/home/tom/illumos-fusefs-Version-1.3.1_funktioniert_kernel_laden_dirty_patch__sonst_ganz_ungetestet/kernel/amd64/Makefile
--- illumos-fusefs-Version-1.3.1/kernel/amd64/Makefile	2016-05-04 21:06:32.000000000 +0000
+++ illumos-fusefs-Version-1.3.1/kernel/amd64/Makefile	2016-11-26 02:35:22.794756446 +0000
@@ -23,7 +23,7 @@
 # Copyright 2008 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
 
-CFLAGS=-xmodel=kernel -m64
+CFLAGS=-xmodel=kernel -m64 -DDONT_CACHE_ATTRIBUTES
 LINTFLAGS = -m64
 include ../Makefile.com
 
@@ -29,6 +29,6 @@
 
 DIRSAMD64 = $(ROOT)/usr/kernel/drv/amd64
 
-install:
+install:	install_common
 	mkdir -p $(DIRSAMD64)
 	$(INSTALL) -f $(ROOT)/usr/kernel/drv/amd64 fuse

diff -ur illumos-fusefs-Version-1.3.1/kernel/fuse_vfsops.c /export/home/tom/illumos-fusefs-Version-1.3.1_funktioniert_kernel_laden_dirty_patch__sonst_ganz_ungetestet/kernel/fuse_vfsops.c
--- illumos-fusefs-Version-1.3.1/kernel/fuse_vfsops.c	2016-05-04 21:06:32.000000000 +0000
+++ illumos-fusefs-Version-1.3.1/kernel/fuse_vfsops.c	2016-11-26 02:50:01.178593514 +0000
@@ -180,8 +180,10 @@
 	vp->v_data = vdata;
 	vdata->nodeid = nodeid;
 	vdata->par_nid = parent_nid;
-	vdata->uid = (uid_t)-1;
-	vdata->gid = (gid_t)-1;
+#ifndef DONT_CACHE_ATTRIBUTES
+	vdata->uid = (uid_t)-1;
+	vdata->gid = (gid_t)-1;
+#endif
 
 	/* Create the list for storing file handles */
 	list_create(&vdata->fh_list, sizeof (fuse_file_handle_t),
diff -ur illumos-fusefs-Version-1.3.1/kernel/fuse_vnops.c /export/home/tom/illumos-fusefs-Version-1.3.1_funktioniert_kernel_laden_dirty_patch__sonst_ganz_ungetestet/kernel/fuse_vnops.c
--- illumos-fusefs-Version-1.3.1/kernel/fuse_vnops.c	2016-11-26 14:30:25.385774335 +0000
+++ illumos-fusefs-Version-1.3.1/kernel/fuse_vnops.c	2016-11-26 02:37:25.641173308 +0000
@@ -1165,7 +1165,9 @@
 		 *	have been modified, so invalidate the cached
 		 *	attributes.
 		 */
+#ifndef DONT_CACHE_ATTRIBUTES
 		invalidate_cached_attrs(vp);
+#endif
 
 		/*
 		 * XXX: Is this required as this might have already been
@@ -1373,8 +1375,7 @@
 		proc_t *p = ttoproc(curthread);
 
 		mutex_enter(&p->p_lock);
-		(void) rctl_action(rctlproc_legacy[RLIMIT_FSIZE], p->p_rctls,
-		    p, RCA_UNSAFE_SIGINFO);
+		(void) rctl_action(RCHD_PROC_LGC_FSIZE, p, RCA_UNSAFE);
 		mutex_exit(&p->p_lock);
 		return (EFBIG);
 	}
--- illumos-fusefs-Version-1.3.1/kernel/Makefile.orig	2016-05-04 21:06:32.000000000 +0000
+++ illumos-fusefs-Version-1.3.1/kernel/Makefile	2016-11-26 18:01:52.545271935 +0000
@@ -41,14 +41,14 @@
 pkg:= TARGET = pkg
 lint:= TARGET = lint
 
-all: $(MACH) $(MACH64)
+all: $(MACH64)
 
-clean: $(MACH) $(MACH64)
+clean: $(MACH) $(MACH64)
 	rm -fr $(DIRS) $(ROOT) $(PKGDIR)
 
-install: preinstall $(MACH) $(MACH64)
+install: preinstall $(MACH64)
 
-lint: $(MACH) $(MACH64)
+lint: $(MACH) $(MACH64)
 
 pkg: prepkg pkgdefs
 
@@ -58,7 +58,7 @@
 prepkg:
 	mkdir -p $(PKGDIR)
 
-$(MACH) $(MACH64) pkgdefs: FRC
+$(MACH64) pkgdefs: FRC
 	@cd $@; pwd; $(MAKE) $(TARGET)
 
 FRC:
--- illumos-fusefs-Version-1.3.1/kernel/Makefile.com.orig       2016-11-27 01:06:56.683548269 +0000
+++ illumos-fusefs-Version-1.3.1/kernel/Makefile.com    2016-11-27 01:08:39.395253566 +0000
@@ -57,7 +57,6 @@
 	$(LINT) $(LLINTFLAGS) $(SRC)
 
 install_common:
-	$(INSTALL) -f $(ROOT)/usr/kernel/drv fuse
 	$(INSTALL) -f $(ROOT)/usr/kernel/drv ../fuse.conf
 
 clean:

